### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by Have-Samuel**
For the PR:
I models should include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.
- Go to `rails c` and check if your methods are working.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,5 +1,16 @@
 class Post < ActiveRecord::Base
-  belong_to :user
+  belongs_to :user, foreign_key: :author_id
   has_many :likes
   has_many :comments
+
+  def update_post_conuter
+    user.increment!(:post_counter)
```
Comment (uwadonat) [inline app/models/post.rb]: - [ ] Let's use Rails Active Callback `after_save` to call `update_post_counter` method every time a new post has been saved to database, then make `update_post_counter` method private. 


`after_save :update_post_counter`

  private

  def update_post_counter
       author.increment!(:posts_counter)
    end

File: app/models/likes.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Like < ActiveRecord::Base
-  belong_to :user
-  belong_to post
+  belongs_to :user, foreign_key: :author_id
+  belongs_to post, foreign_key: :post_id
+
+  def update_likes_conuter
+    post.increment!(:likes_counter)
+  end
```
Comment (uwadonat) [inline app/models/likes.rb]: - [ ] Let's use Rails Active Callback `after_save` to call `update_likes_conuter` method.

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @Have-Samuel,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your commit messages are very descriptive ✔️
- No linter errors ✔️
- Descriptive pull request message ✔️
almost

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @Have-Samuel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**